bigtable/admin/table_admin_test.cc

```diff
@@ -132,6 +137,27 @@ TEST_F(TableAdminTest, ListTablesRecoverableFailures) {
   ASSERT_EQ(4UL, actual.size());
   EXPECT_EQ(instance_name + "/tables/t0", actual[0].name());
   EXPECT_EQ(instance_name + "/tables/t1", actual[1].name());
-  EXPECT_EQ(instance_name + "/tables/t0", actual[2].name());
-  EXPECT_EQ(instance_name + "/tables/t1", actual[3].name());
+  EXPECT_EQ(instance_name + "/tables/t2", actual[2].name());
+  EXPECT_EQ(instance_name + "/tables/t3", actual[3].name());
+}
+
+/// @test Verify that bigtable::TableAdmin::ListTables handles unrecoverable
+/// failures.
+TEST_F(TableAdminTest, ListTablesUnrecoverableFailures) {
+  using namespace ::testing;
+
+  bigtable::TableAdmin tested(client_, "the-instance");
+  auto mock_unrecoverable_failure = [](
+      grpc::ClientContext* ctx, btproto::ListTablesRequest const& request,
+      btproto::ListTablesResponse* response) {
+    return grpc::Status(grpc::StatusCode::PERMISSION_DENIED, "uh oh");
+  };
+  EXPECT_CALL(*table_admin_stub_, ListTables(_, _, _))
+      .WillOnce(Invoke(mock_unrecoverable_failure));
+  // We expect the TableAdmin to make 5 calls and to let the client know about
```

Comment:
The "5" in this line is in stark contrast to the "1" on the next line — should this be updated?